### PR TITLE
Add Terraform Config for Feature Matrix Creation Cloud Function

### PR DESCRIPTION
**Pull request by waltaskew**
- Add script cp_cloud_function_files use to prepare the .zip file containing cloud function source code.
- Add terraform config for a cloud function which builds feature tensors for chunks on writes to the chunk bucket.

### CODE REVIEW

File: terraform/cp_cloud_function_files.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+set -e
```
Comment (waltaskew) [inline terraform/cp_cloud_function_files.sh:1]: Do folks have thoughts on how to integrate this more with the terraform? Right now, my process is to do
```
./cp_cloud_function_files.sh && terraform apply
```
to deploy new code to the cloud functions. It'd be rad if it was just `terraform apply`

**Comment by alevenberg** [comment]
Have you tested apply/plan/destroy on a project with this config? If so, do you have instructions for what commands you use?

**Comment by waltaskew** [comment]
> Have you tested apply/plan/destroy on a project with this config? If so, do you have instructions for what commands you use?

commands are
`./cp_cloud_function_files.sh && terraform apply`
I left a comment in here -- curious if folks have suggestions for folding in that shell script so it's just a `terraform apply`

the infrastructure described in here is up and running!

**Comment by alevenberg** [comment]
> > Have you tested apply/plan/destroy on a project with this config? If so, do you have instructions for what commands you use?
> 
> commands are `./cp_cloud_function_files.sh && terraform apply` I left a comment in here -- curious if folks have suggestions for folding in that shell script so it's just a `terraform apply`
> 
> the infrastructure described in here is up and running!

It seems strange to me that you would need both. I think you are copying files into a remote destination (bucket). You can upload a file in terraform or point to a docker image for the code (https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/cloudfunctions_function) (i think)

**Comment by waltaskew** [comment]
> > > Have you tested apply/plan/destroy on a project with this config? If so, do you have instructions for what commands you use?
> > 
> > 
> > commands are `./cp_cloud_function_files.sh && terraform apply` I left a comment in here -- curious if folks have suggestions for folding in that shell script so it's just a `terraform apply`
> > the infrastructure described in here is up and running!
> 
> It seems strange to me that you would need both. I think you are copying files into a remote destination (bucket). You can upload a file in terraform or point to a docker image for the code (https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/cloudfunctions_function) (i think)

`cp_cloud_function_files.sh` does not copy files into GCS. It moves files into a local cloud_function_source directory. From there, teraform is responsible for zipping the contents of that directory and uploading it to GCS.

File: terraform/cp_cloud_function_files.sh (hunk 1 above)
Comment (alevenberg) [reply terraform/cp_cloud_function_files.sh:1]: I wonder if you can use `repo_source` somehow. I don't know much about cloud functions though. https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/cloudfunctions2_function#repo_source

**Review by prashantkul — APPROVED** [review]

File: terraform/cp_cloud_function_files.sh (hunk 1 above)
Comment (waltaskew) [reply terraform/cp_cloud_function_files.sh:1]: We aren't working with Cloud Source Repositories, so I don't think we can use `repo_source`

Comment (alevenberg) [reply terraform/cp_cloud_function_files.sh:1]: I'm fine with you submitting as is-- I don't know much about cloud functions, but it seems strange that we have to run a script before `terraform apply`. But we can change this later on too.

Comment (waltaskew) [reply terraform/cp_cloud_function_files.sh:1]: > I'm fine with you submitting as is-- I don't know much about cloud functions, but it seems strange that we have to run a script before `terraform apply`. But we can change this later on too.

I specifically asked if folks had ideas on how to get rid of that `cp_cloud_function_files.sh` script and incorporate it into terraform!

`data "archive_file"` is a bit restrictive. The script is just copying files to make them easier for terraform to pull into a zip via `data "archive_file"`

**Pull request merged**